docs/plugins/pluggable_authentication.md

```diff
@@ -243,3 +435,93 @@ The response payload from the server will minimally contain the following:
 ```
 
 The `request_result` key contains a time-limited, randomly-generated iRODS password. This password should be used to authenticate using the native authentication scheme. How the client implements this authentication is entirely up to the developer. For reference, the C++ `pam_password` client-side authentication plugin saves the randomly-generated password to a `.irodsA` file and then starts the authentication process from the "start" operation using the native authentication scheme and the password in that `.irodsA` file.
+
+#### irods authentication
+
+The `irods` authentication plugin has three server-side operations which must be called in order. The inputs and outputs from each are described below.
+
+**Step 1: `server_prepare_auth_check`**
+
+This operation - like all server-side plugin operations - requires a "scheme" in the request payload so that the server knows which authentication plugin to load. Otherwise, this operation has no requirements in terms of keys to include in the request payload. The request payload should minimally contain the following:
+```json
+{
+    "scheme": "irods"
+}
+```
+This step enforces secure communications between client and server before any sensitive information (i.e. passwords) are sent over the network.
+
+The response payload should be identical to the request payload.
+
+**Step 2: `server_auth_with_password`**
+
+This operation requires 3 keys: `user_name`, `zone_name`, and `password`.
+
+`user_name` and `zone_name` are the username and zone name for the user being authenticated. A user of the specified name belonging to the specified zone must exist in the local zone's catalog in order for authentication to succeed.
+
+`password` is the user's **plaintext password**. This is the main reason that TLS is required when using this authentication scheme.
+
+How the plugin gets the user's password is entirely up to the library developer. For reference, the C++ client-side authentication plugin has two ways of obtaining the password:
+
+- The `password` key in the request payload
+- Prompting the user and receiving a password from `stdin`
+
+The request payload should minimally contain the following:
+```json
+{
+    "scheme": "irods",
+    "user_name": "<user name>",
+    "zone_name": "<zone name>",
+    "password": "<user password>"
```

Comment:
Consider using the types to match section about session_token_expires.


```suggestion
    "user_name": "<string>",
    "zone_name": "<string>",
    "password": "<string>"
```

Reply:
Done